src/components/UserInfo.jsx

```diff
@@ -6,12 +6,47 @@ import StyledUserInfo from './styledComponents/SuserInfo';
 
 function UserInfo() {
   const [user, setUser] = useState([]);
+  const [activity, setActivity] = useState([]);
+  const [formActivity, setFormActivity] = useState([]);
   const { login } = useParams();
+
+  // Function calculat the number of days since the last connection
+  function calculatActivity(data) {
+    const saneData = { ...data };
+    saneData.updated_at = data.updated_at;
+    const userDateConnect = saneData.updated_at; // Date of the last connection
+    const miliUserDateConnect = Date.parse(userDateConnect); // Time in miliseconds between the last connection and 01-01-1970
+    const dateOfDay = new Date(); // Date of the day
+    const miliDateOfDay = Date.parse(dateOfDay); //  Time in milliseconds between today and 01-01-1970
+    const calOfDay =
+      (miliDateOfDay - miliUserDateConnect) / (60 * 60 * 24 * 1000); // calculate the time elapsed between the last connection and today in days
+    return calOfDay;
+  }
+
+  function accountCreationDate(data) {
+    const saneData = { ...data };
+    saneData.created_at = data.created_at.substring(0, 10);
+    return saneData;
+  }
+  function formatActivity() {
+    if (activity <= 7) return 'activityhight';
+    if (activity <= 14) return 'activitymedium';
+    if (activity <= 60) return 'activitylow';
+    return 'activitynul';
+  }
   useEffect(() => {
     axios.get(`https://api.github.com/users/${login}`).then(({ data }) => {
-      setUser(data);
+      const userActivity = calculatActivity(data);
+      setActivity(userActivity);
+      const saneData = accountCreationDate(data);
+      setUser(saneData);
+      const pictureActivity = formatActivity(data);
+      setFormActivity(pictureActivity);
+      const image = document.querySelector('.imgActivity');
+      image.src = `../img/${formActivity}.png`;
```

Comment:
Wait, what !?
Up to now I was very proud of you <3
Why don't you do this directly in the render of this component ? Something like `<img src=`../img/${formatActivity}.png` />` ?